state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -122,6 +123,15 @@ pub enum DepositIgnoredReason {
 	NotEnoughToPayFees,
 }
 
+/// Holds information about a tainted transaction.
+#[derive(RuntimeDebug, Eq, PartialEq, Clone, Encode, Decode, TypeInfo)]
+pub struct TaintedTransactionDetails<AccountId> {
+	/// The broker that created the tainted transaction.
+	pub broker: AccountId,
+	/// The address we use for refunding.
+	pub refund_address: Option<ForeignChainAddress>,
```

Comment:
I would have thought that having the refund address here was unnecessary since we already have channel refund params.

Reply:
We take the refund address from the deposit channel. But at the same point when we later want to refund we need this address, right? The other option would be saving the channel_id, but saving the refund address directly seemed a bit more convenient to me.

Reply:
See my other comment - we actually need more than just the refund address so that we can construct a refund tx. I agree we should save it though, rather than just store a reference to the channel (unless we can guarantee that the channel won't be deleted).